File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    w = set()
    w.add(words[0])
    
    for i in range(1, len(words)):
        if words[i] in w or words[i-1][-1] != words[i][0]:
            return [i % n + 1, i // n + 1]
        w.add(words[i])
        
    return [0, 0]

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        Set<String> pre = new HashSet<>();
        pre.add(words[0]);
        
        for (int i = 1; i < words.length; i++) {
            if (pre.contains(words[i]) || words[i-1].charAt(words[i-1].length() - 1)  != words[i].charAt(0)) {
                return new int[] {i % n + 1, i / n + 1};
            }
            pre.add(words[i]);
        }
        
        return new int[]{0,0};
    }
}